src/FESpaces.jl

```diff
@@ -33,8 +37,16 @@ function FESpaces.zero_free_values(f::DistributedFESpace)
   fill!(vec,zero(eltype(vec)))
 end
 
+function FESpaces.zero_dirichlet_values(f::DistributedFESpace)
+  V = get_vector_type(f)
+  vec = allocate_vector(V,num_dirichlet_dofs(f))
+  fill!(vec,zero(eltype(vec)))
+end
+
 FESpaces.num_free_dofs(f::DistributedFESpace) = length(get_free_dof_ids(f))
 
+FESpaces.num_dirichlet_dofs(f::DistributedFESpace) = length(get_dirichlet_dof_ids(f))
```

Comment:
I would say that even this one needs to return an `AbstractPData` object.

Reply:
Removed in fa9873ea6a8b6bebe8a481016138297fc0687dfc